### PR TITLE
[bitnami/grafana-loki] Implement query-scheduler configuration

### PR DESCRIPTION
**Pull request by bcarlock-emerge**
### Description of the change

Implementation of the grafana-loki query-scheduler and corresponding changes to query-frontend and querier.
https://grafana.com/docs/loki/latest/operations/scalability/#separate-query-scheduler
https://github.com/grafana/loki/blob/main/production/ksonnet/loki/query-scheduler.libsonnet

### Benefits

Allows query-frontend & querier to scale more gracefully, increase performance. 
"The Query frontend has an in-memory queue that can be moved out into a separate process similar to the [Cortex Query Scheduler](https://cortexmetrics.io/docs/operations/scaling-query-frontend/#query-scheduler). This allows running multiple query frontends." https://grafana.com/docs/loki/latest/operations/scalability/#separate-query-scheduler

### Possible drawbacks

N/A- as far as I'm aware

### Applicable issues

N/A

### Additional information

Tests have been run and compared against a running manual configuration with all expected values returned. 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-loki/Chart.yaml [hunk 1]
```diff
@@ -44,4 +44,4 @@ name: grafana-loki
 sources:
   - https://github.com/bitnami/bitnami-docker-grafana-loki
   - https://github.com/grafana/loki/
-version: 2.1.2
+version: 2.2.2
```
Comment (fmulero) [inline bitnami/grafana-loki/Chart.yaml:47]: I think there is a typo here:
```suggestion
version: 2.2.0
```

File: bitnami/grafana-loki/templates/query-scheduler/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  minReadySeconds: 10
+  replicas: {{ .Values.queryScheduler.replicaCount }}
+  {{- if .Values.queryScheduler.updateStrategy }}
+  strategy: {{- toYaml .Values.queryScheduler.updateStrategy | nindent 4 }}
+  {{- end }}
+  revisionHistoryLimit: 2
```
Comment (fmulero) [inline bitnami/grafana-loki/templates/query-scheduler/deployment.yaml:22]: `revisionHistoryLimit` is not used in other deployments. We should keep consistency between all the deployments in this chart. Could you remove it?

File: bitnami/grafana-loki/templates/query-scheduler/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    app.kubernetes.io/part-of: grafana-loki
+    app.kubernetes.io/component: query-scheduler
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  minReadySeconds: 10
```
Comment (fmulero) [inline bitnami/grafana-loki/templates/query-scheduler/deployment.yaml:17]: Is there any requirement or reason to set `minReadySeconds`? If there is a reason to use it, could make sense to create a value that allow users change it

File: bitnami/grafana-loki/values.yaml [hunk 4]
```diff
@@ -200,9 +200,10 @@ loki:
             max_size_items: 1024
             validity: 24h
           {{- end }}
-
+    {{- if not .Values.queryScheduler.enabled }}
```
Comment (fmulero) [inline bitnami/grafana-loki/values.yaml:203]: This is affecting to all grafana-loki componente, not only the querier. Is it expected?

**Review by fmulero — CHANGES_REQUESTED** [review]
Great @bcarlock-emerge, nice work! Could you have a look to the comments?

File: bitnami/grafana-loki/templates/query-scheduler/deployment.yaml (hunk 3 above)
Comment (bcarlock-emerge) [reply bitnami/grafana-loki/templates/query-scheduler/deployment.yaml:17]: I'm honestly not sure, it was generated by Tanka. I'll add it as a value.

**Comment by fmulero** [comment]
Hi @bcarlock-emerge, there are too many changes related to other charts, not sure if you had any problem during merge. Could you resolve conflicts and check if the changes were done intentionally?

**Comment by bcarlock-emerge** [comment]
> Hi @bcarlock-emerge, there are too many changes related to other charts, not sure if you had any problem during merge. Could you resolve conflicts and check if the changes were done intentionally?

Sorry for the delay, I've addressed the issues with other charts.

**Pull request merged**